Use a small snap to node distance when Projecting candidates

dnesbitt61 opened the pull request:
# Issue

fixes #3230

Use a node snapping distance of 1 meter in calls to helpers::Project. This fixes an issue where an incorrect turn followed by immediate Uturn occurs at the node.

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

kevinkreiser commented:
yeah i vaguely remember removing node snapping for map matching because it causes other problems, do all the tests pass?

dnesbitt61 commented:
tests pass. I used a very short distance (1 meter) to hopefully limit any side effects.

kevinkreiser commented:
yeah sounds like it could be floating point noise maybe. perhaps even a smaller one would do the same trick

kevinkreiser commented:
since this is kind of important would you mind adding a changelog entry?

dnesbitt61 commented:
added ChangeLog entry. I tried with 0.5 meter node snapping threshold and this also fixes the 2 test routes. I know the one I added yesterday had a distance of about 0.16 meters.
